tools-v2/pkg/cli/command/curvebs/update/peer/peer.go

```diff
@@ -0,0 +1,196 @@
+}
+
+func (rRpc *ResetRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	rRpc.Client = cli2.NewCliService2Client(cc)
+}
+
+func (rRpc *ResetRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return rRpc.Client.ResetPeer(ctx, rRpc.Request)
+}
+
+// Options rpc request options.
+type Options struct {
+	Timeout    time.Duration
+	RetryTimes int32
+}
+
+type PeerCommand struct {
+	basecmd.FinalCurveCmd
+
+	// request parameters
+	opts          Options
+	logicalPoolID uint32
+	copysetID     uint32
+
+	resetPeer *common.Peer
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*PeerCommand)(nil) // check interface
+
+// NewCommand ...
+func NewPeerCommand() *cobra.Command {
+	peerCmd := &PeerCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "peer",
+			Short:   "update(reset) the peer from the copyset",
+			Example: updateExample,
+		},
+	}
+	basecmd.NewFinalCurveCli(&peerCmd.FinalCurveCmd, peerCmd)
+	return peerCmd.Cmd
+}
+
+func (pCmd *PeerCommand) AddFlags() {
+	config.AddRpcRetryTimesFlag(pCmd.Cmd)
+	config.AddRpcTimeoutFlag(pCmd.Cmd)
+
+	config.AddBSLogicalPoolIdFlag(pCmd.Cmd)
+	config.AddBSCopysetIdFlag(pCmd.Cmd)
+}
+
+// ParsePeer parse the peer string
+func ParsePeer(peer string) (*common.Peer, *cmderror.CmdError) {
+	cs := strings.Split(peer, ":")
+	if len(cs) != 3 {
+		pErr := cmderror.ErrGetPeer()
+		pErr.Format(fmt.Sprintf("error format for the peer info %s", peer))
+		return nil, pErr
+	}
+	id, err := strconv.ParseUint(cs[2], 10, 64)
+	if err != nil {
+		pErr := cmderror.ErrGetPeer()
+		pErr.Format(fmt.Sprintf("error format for the peer id %s", cs[2]))
+		return nil, pErr
+	}
+	cs = cs[:2]
+	address := strings.Join(cs, ":")
+	return &common.Peer{
+		Id:      &id,
+		Address: &address,
+	}, nil
+}
+
+func (pCmd *PeerCommand) Init(cmd *cobra.Command, args []string) error {
+	pCmd.opts = Options{}
+
+	var err error
+	var e *cmderror.CmdError
+
+	pCmd.opts.Timeout = config.GetFlagDuration(pCmd.Cmd, config.RPCTIMEOUT)
+	pCmd.opts.RetryTimes = config.GetFlagInt32(pCmd.Cmd, config.RPCRETRYTIMES)
+
+	pCmd.copysetID, err = config.GetBsFlagUint32(pCmd.Cmd, config.CURVEBS_COPYSET_ID)
+	if err != nil {
+		return err
+	}
+	pCmd.logicalPoolID, err = config.GetBsFlagUint32(pCmd.Cmd, config.CURVEBS_LOGIC_POOL_ID)
+	if err != nil {
+		return err
+	}
+
+	pCmd.resetPeer, e = ParsePeer(args[0])
+	if e != nil {
+		return e.ToError()
+	}
+	return nil
+}
+
+func (pCmd *PeerCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&pCmd.FinalCurveCmd, pCmd)
+}
+
+func (pCmd *PeerCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	out := make(map[string]string)
+	out[cobrautil.ROW_PEER] = fmt.Sprintf("%s:%d", pCmd.resetPeer.GetAddress(), pCmd.resetPeer.GetId())
+	out[cobrautil.ROW_COPYSET] = fmt.Sprintf("(%d:%d)", pCmd.logicalPoolID, pCmd.copysetID)
+
+	var err *cmderror.CmdError
+	pCmd.Result, err = pCmd.execResetPeer()
+	if err != nil {
+		out[cobrautil.ROW_RESULT] = "failed"
```

Comment:
Such commonly used values can be found in internal/utils
Defined as a constant in /row.go